listenbrainz/webserver/views/api_tools.py

```diff
@@ -121,9 +121,29 @@ def _send_listens_to_queue(listen_type, listens):
         )
 
 
+def _raise_error_if_has_unicode_null(value, listen):
+    if isinstance(value, str) and '\x00' in value:
+        raise APIBadRequest("{} contains a unicode null".format(value), listen)
+
+
+def check_for_unicode_null_recursively(listen: Dict):
```

Comment:
Alternative way is to first call `flatten_dict` and then check for nulls. That was the way it is done in `listen.py`. I think either way is fine but didn't use that currently because it doesn't handle lists.